### PR TITLE
Unfreeze werkzeug

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Hm, the tests are actually passing on my 3.9, but I guess Flask-Markdown is pretty old. I'll give Flask-Misaka a spin!

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Everything needed in production
-werkzeug<2.3
+werkzeug>=3.0.3
 Flask>=2.0
 Flask-Bootstrap
 Flask-CORS
 path.py
-Flask-Markdown
+Flask-Misaka
```
Comment (northwestwitch) [inline requirements.txt:7]: 🇯🇵

Comment (dnil) [reply requirements.txt:7]: It looked markupish enough I think:
Screenshot 2024-05-07 at 09 13 03

File: requirements.txt [hunk 2]
```diff
@@ -29,7 +29,7 @@ importlib_resources
 # webapp login
 authlib
 flask_login
-flask-ldapconn
+flask-ldapconn @ git+https://github.com/Clinical-Genomics/flask-ldapconn.git@v0.10.2
```
Comment (northwestwitch) [inline requirements.txt:32]: I have a feeling that they merged my PR on the original repo without testing it, so I'm a bit worried that this change might brake the LDAP connection functionality.. I think we should at least trying the LDAP connection following the simple Futurama tutorial?

**Review by northwestwitch — COMMENTED** [review]
See my comment about the LDAP connection. Could we test it before merging?

File: requirements.txt (hunk 2 above)
Comment (dnil) [reply requirements.txt:32]: 💯 Sure, I was certain you had tested it before PRing an external repo.

Comment (northwestwitch) [reply requirements.txt:32]: I've tried to use the original tests of that repo, but I couldn't make it work

File: requirements.txt (hunk 1 above)
Comment (dnil) [reply requirements.txt]: And so does Python-Markdown with a simple jinja filter wrap:
Screenshot 2024-05-07 at 15 18 57

**Comment by northwestwitch** [comment]
Login with LDAP still works:

image

File: .github/workflows/tests_and_cov.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@ jobs:
 
       - name: Install deps
         run: |
-          pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -e .
+          pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -r requirements.txt -e .
```
Comment (northwestwitch) [inline .github/workflows/tests_and_cov.yml:24]: What changed here?

File: .github/workflows/tests_and_cov.yml [hunk 4]
```diff
@@ -57,7 +57,7 @@ jobs:
           key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements-dev.txt') }}
 
       - name: Install the HTML 2 PDF renderer
-        run: sudo apt-get update || true && sudo apt-get install -y wkhtmltopdf
+        run: sudo apt-get update || true && sudo apt-get install -y wkhtmltopdf git
```
Comment (northwestwitch) [inline .github/workflows/tests_and_cov.yml:60]: why?

File: requirements.txt (hunk 2 above)
Comment (northwestwitch) [reply requirements.txt:32]: 👍🏻 but let's open an issue so we remember to switch to the original repo PyPi once they update

**Review by northwestwitch — APPROVED** [review]
Nice! 💯

File: .github/workflows/tests_and_cov.yml (hunk 3 above)
Comment (dnil) [reply .github/workflows/tests_and_cov.yml:24]: Let's see, I think it didn't try to install the git+http from the requirements without giving it explicitly?

File: .github/workflows/tests_and_cov.yml (hunk 4 above)
Comment (dnil) [reply .github/workflows/tests_and_cov.yml:60]: git does not seem to be installed on the tester image for some reason?

Comment (dnil) [reply .github/workflows/tests_and_cov.yml:60]: All of those explicit `git` installs can be removed as soon as we can install `flask-ldapconn` from pypi again.

File: .github/workflows/tests_and_cov.yml (hunk 3 above)
Comment (northwestwitch) [reply .github/workflows/tests_and_cov.yml:24]: Ah yes, right

File: .github/workflows/tests_and_cov.yml (hunk 4 above)
Comment (northwestwitch) [reply .github/workflows/tests_and_cov.yml:60]: Understood! 👍🏻

**Pull request merged**